torch/csrc/jit/passes/autocast.cpp

```diff
@@ -446,9 +457,12 @@ bool autocastEnabled() {
 void Autocast(const std::shared_ptr<Graph>& graph) {
   GRAPH_DUMP("\nBefore Autocast: ", graph);
   if (autocastEnabled()) {
+    auto is_gpu_enabled = at::autocast::is_enabled();
+    auto is_cpu_enabled = at::autocast::is_cpu_enabled();
     AutocastContext init = {
-        at::autocast::is_enabled(),
-        at::autocast::is_cpu_enabled(),
+        is_gpu_enabled,
+        is_cpu_enabled,
+        is_gpu_enabled || is_cpu_enabled,
```

Comment:
This should just be always true, since we are reflecting if global state is set appropriately, not whether it is set to true